ldpd/lde.c

```diff
@@ -1708,6 +1721,38 @@ on_get_label_chunk_response(uint32_t start, uint32_t end)
 		current_label_chunk = listtail(label_chunk_list);
 }
 
+void
+lde_free_label(uint32_t label)
+{
+	struct listnode	*node;
+	struct label_chunk	*label_chunk;
+	uint64_t		pos;
+
+	node = listhead(label_chunk_list);
+
+	while (node) {
+		label_chunk = listgetdata(node);
+
+		if (label_chunk != NULL)
+			if ((label < label_chunk->end) && (label >= label_chunk->start)) {
```

Comment:
Should this be `<= label_chunk->end`? The code in `lde_get_next_label()` looks as if it will use all of the range, including the `end` value. It looks as if this will fail to find the matching chunk for the last label value in the chunk. Were you able to test this and see that using then freeing the last label in a chunk will trigger the new 'release' code ?

Reply:
Actually,  there was another bug in next line.  Since my cross compiler did not show  -wshift_count_overflow, I did not notice this bug. Because of this bug,  the bug you mentioned was not found in my previous testing.  I  will test again and commit code tomorrow. 
Thanks
Binu

-                       if ((label < label_chunk->end) && (label >= label_chunk->start)) {
-                               pos = 1 << (label - label_chunk->start);
+                       if ((label <= label_chunk->end) && (label >= label_chunk->start)) {
+                               pos = 1ULL << (label - label_chunk->start);